service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/PaymentServiceTest.java

```diff
@@ -99,6 +103,52 @@ void setup() {
             when(feeService.getFeesDataForC2(WITHOUT_NOTICE)).thenReturn(buildFeesData(feeForC2WithoutNotice));
         }
 
+        @Test
+        void shouldRetryPaymentsApiWhenInternalServerErrorThrown() {
+            CreditAccountPaymentRequest expectedPaymentRequest = testCreditAccountPaymentRequestBuilder()
+                .customerReference("1")
+                .amount(feeForC2WithNotice.getCalculatedAmount())
+                .fees(List.of(feeForC2WithNotice))
+                .build();
+
+            when(paymentApi.createCreditAccountPayment(AUTH_TOKEN, SERVICE_AUTH_TOKEN, expectedPaymentRequest))
+                .thenThrow(FeignException.InternalServerError.class);
+
+            assertThrows(Exception.class, () -> paymentService.callPaymentsApi(expectedPaymentRequest));
+            verify(paymentApi, times(3)).createCreditAccountPayment(any(), any(), any());
```

Comment:
minor: personally i would do 
verify(paymentApi, times(3)).createCreditAccountPayment(AUTH_TOKEN, SERVICE_AUTH_TOKEN, expectedPaymentRequest);

to prove it was called with same params